File: Services/MooMed.Stateful.SearchService/Service/SearchService.cs
using System.Threading.Tasks;
using JetBrains.Annotations;
using MooMed.Common.Definitions.Logging;
using MooMed.Common.Definitions.Models.Search;
using MooMed.Common.ServiceBase.ServiceBase;
using MooMed.Core.DataTypes;
using MooMed.ServiceBase.Services.Interface;

namespace MooMed.Stateful.SearchService.Service
{
	public class SearchService : MooMedServiceBaseWithLogger, ISearchService
	{
		[NotNull]
		private readonly IAccountService _accountService;

		public SearchService(
			[NotNull] IMooMedLogger logger,
			[NotNull] IAccountService accountService)
			: base(logger)
		{
			_accountService = accountService;
		}

		[ItemNotNull]
		public async Task<ServiceResponse<SearchResult>> Search([NotNull] string query)
		{
			var accountsStartingWithNameResponse = await _accountService.FindAccountsStartingWithName(query);

			var searchResult = new SearchResult
			{
				CorrespondingUsers = accountsStartingWithNameResponse.PayloadOrNull,
			};

			return ServiceResponse<SearchResult>.Success(searchResult);
		}
	}
}

File: Services/MooMed/React/Modules/Search/Service/SearchService.ts
// Functionality
import { ISearchService } from "Definitions/Service";
import * as searchCommunication from "modules/Search/Communication/SearchCommunication";
import { SearchResult } from "modules/Search/types";

export default class SearchService implements ISearchService {

	public async search(query: string): Promise<SearchResult> {
		const searchResponse = await searchCommunication.search({
			query
		});

		return {
			correspondingAccounts: searchResponse.payload.correspondingUsers
		};
	}
}